tools/building.py

```diff
@@ -1089,8 +1089,16 @@ def add_to_path(dirname):
     outfile = filename + '.cc.js'
     configuration.get_temp_files().note(outfile)
 
-    args = ['--compilation_level', 'ADVANCED_OPTIMIZATIONS' if advanced else 'SIMPLE_OPTIMIZATIONS',
-            '--language_in', 'ECMASCRIPT5']
+    args = ['--compilation_level', 'ADVANCED_OPTIMIZATIONS' if advanced else 'SIMPLE_OPTIMIZATIONS']
+    # Keep in sync with ecmaVersion in tools/acorn-optimizer.js
+    args += ['--language_in', 'ECMASCRIPT_2018']
+    # Tell closure not to do any tranpiling or inject any polyfills.
+    # At some point we may want to look into using this as way to convert to ES5 but
+    # babel is perhaps a better tool for that.
+    args += ['--language_out', 'NO_TRANSPILE']
```

Comment:
I am skeptical about what this combo of in and out language will produce. Is `--language_in ECMASCRIPT_2018 --language_out NO_TRANSPILE` guaranteed that Closure will not add ES 2018 constructs of its own to the build output, even if the input did not have any?

If there is no such guarantee, this change means that we'll be upping the minimum requirement from ES5 to ES2018. (in my mind either `--language_out NO_TRANSPILE` is just a convenience shorthand for `--language_out WHATEVER_LANGUAGE_IN_WAS`, or Closure's minification transformation rules are extremely conservative/weak to never leverage anything but lowest ES3 when it does transformations)

Posted https://github.com/google/closure-compiler/issues/3679 to ask to clarify the meaning of the flags.

Reply:
I indent of this PR is certainly not to change the minimum requirement from ES5.   That is a separate issue.

The intent is simply to allow the user code in EM_JS and EM_ASM and pre-js/post-js to pass through our pipeline without erroring out when running closure.

I share you concern though, it would be really bad if closure added any ES6 constructs here.

Perhaps the safest thing to do would be to verify the output using some kind of ES version checker.  It looks like such things do exists: https://github.com/dollarshaveclub/es-check.    What if the test suite ran this checker by default on all emscripten output?  Would that be enough empirical evidence prove this is doing why we think its doing?     Adding this would also prevent emscripten developers to sneaking in ES6 constructs, which I believe we have no protections against today outside of the review system.

Reply:
I run the entire wasm2 and other test suites with #12139 to verify that this doesn't generate any non-ES5 output.

Reply:
#12139 has now landed so we know for sure we are not introducing any ES6.. and if we update closure and they start added any we will know right away.